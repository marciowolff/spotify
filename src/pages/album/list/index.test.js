import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import List from './index';

const dataList = {
  tracks: {
    items: [
      { id: 0, track_number: 1, name: 'name track 1', duration_ms: 10000 },
      { id: 1, track_number: 2, name: 'name track 2', duration_ms: 20000 },
    ],
  },
};

it('List should be a function', () => {
  expect(typeof List).toBe('function');
});

it('Should List to match snapshot', () => {
  const tree = renderer.create(<List data={dataList} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Should List with parameter data', () => {
  const wrapper = shallow(<List data={dataList} />);

  expect(wrapper.find('ul').hasClass('list')).toBeTruthy();
  expect(wrapper.find('li')).toHaveLength(2);
});

it('Should List without parameter data', () => {
  const wrapper = shallow(<List />);

  expect(wrapper.find('ul').hasClass('list')).toBeTruthy();
  expect(wrapper.find('li')).toHaveLength(0);
});

it('Should List with parameer data equal null', () => {
  const wrapper = shallow(<List data={null} />);

  expect(wrapper.find('ul').hasClass('list')).toBeTruthy();
  expect(wrapper.find('li')).toHaveLength(0);
});

it('Should List with parameer data equal undefined', () => {
  const wrapper = shallow(<List data={undefined} />);

  expect(wrapper.find('ul').hasClass('list')).toBeTruthy();
  expect(wrapper.find('li')).toHaveLength(0);
});
