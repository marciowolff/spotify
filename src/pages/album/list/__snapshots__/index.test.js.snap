// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should List to match snapshot 1`] = `
<ul
  className="list"
>
  <li>
    <span
      className="number"
    >
      1
      .
    </span>
    <span
      className="name"
    >
      name track 1
    </span>
    <span
      className="duration"
    >
      10000
    </span>
  </li>
  <li>
    <span
      className="number"
    >
      2
      .
    </span>
    <span
      className="name"
    >
      name track 2
    </span>
    <span
      className="duration"
    >
      20000
    </span>
  </li>
</ul>
`;
